tools/codegen/src/generator/languages/rust/getters.rs

```diff
@@ -63,15 +63,19 @@ impl ImplGetters for ast::Union {
             let getter_stmt = quote!(&self.as_slice()[molecule::NUMBER_SIZE..]);
             (getter_ret, getter_stmt)
         };
-        let match_stmts = self.items().iter().enumerate().map(|(index, inner)| {
-            let item_id = usize_lit(index);
-            let inner = if is_entity {
-                entity_name(inner.typ().name())
-            } else {
-                reader_name(inner.typ().name())
-            };
-            quote!(#item_id => #inner::new_unchecked(inner).into(),)
-        });
+        let match_stmts =
+            self.items()
+                .iter()
+                .map(|item| (item.id(), item))
+                .map(|(index, inner)| {
+                    let item_id = usize_lit(index);
+                    let inner = if is_entity {
+                        entity_name(inner.typ().name())
+                    } else {
+                        reader_name(inner.typ().name())
+                    };
+                    quote!(#item_id => #inner::new_unchecked(inner).into(),)
```

Comment:
no need to double map here:

```suggestion
                .map(|item| {
                    let item_id = usize_lit(item.id());
                    let inner = if is_entity {
                        entity_name(item.typ().name())
                    } else {
                        reader_name(item.typ().name())
                    };
                    quote!(#item_id => #inner::new_unchecked(inner).into(),)
                });
```